src/ConnectedMode/ServerSentEvents/SSESessionFactory.cs

```diff
@@ -200,6 +207,8 @@ public void Dispose()
                     return;
                 }
 
+                onSessionFailedCallback(this).Forget();
```

Comment:
It seems a bit odd to be calling "on failed" from within `Dispose`.
Presumably there are non-failure cases in which the session will be disposed/

It might just be a naming issue - it might be less confusing if the delegate were called `OnSessionDisposedCallback`.

Reply:
Hmm I see what you mean, now that I think about it wouldn't it instead make more sense to move the callback handling into the catch block? Do we actually want it to happen in the dispose?

Reply:
@bigfluffycookie I think originally it was in the try-catch, and for some reason I moved it to the dispose. Was a mistake.